File: data/recomposed_projects/deepseek-coder-33b-instruct/body/commons-graph/evosuite-test/src/test/org/apache/commons/graph/weight/primitive/BigDecimalWeightBaseOperations_ESTest.py
from __future__ import annotations
import math
import time
import re
import os
import decimal
import unittest
import pytest
import io
import unittest
from src.main.org.apache.commons.graph.weight.primitive.BigDecimalWeightBaseOperations import *

# from src.test.org.apache.commons.graph.weight.primitive.BigDecimalWeightBaseOperations_ESTest_scaffolding import *
# from src.main.org.evosuite.runtime.EvoAssertions import *
# from src.main.org.evosuite.runtime.EvoRunnerParameters import *


class BigDecimalWeightBaseOperations_ESTest(unittest.TestCase):

    def test13(self) -> None:

        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()
        bigDecimal0 = decimal.Decimal(10)
        int0 = bigDecimalWeightBaseOperations0.compare(bigDecimal0, bigDecimal0)
        self.assertEqual(0, int0)

    def test12(self) -> None:

        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()

        try:
            bigDecimalWeightBaseOperations0.inverse(None)
            self.fail("Expecting exception: RuntimeError")

        except Exception as e:
            self.assertEqual(str(e), "Expecting exception: RuntimeError")
            self.assertEqual(type(e), RuntimeError)

    def test11(self) -> None:

        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()
        bigDecimal0 = bigDecimalWeightBaseOperations0.identity()
        bigDecimal1 = bigDecimalWeightBaseOperations0.append(bigDecimal0, bigDecimal0)
        self.assertEqual(bigDecimal0, bigDecimal1)

    def test10(self) -> None:

        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()
        bigDecimal0 = bigDecimalWeightBaseOperations0.append(None, None)
        self.assertIsNone(bigDecimal0)

    def test09(self) -> None:

        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()
        bigDecimal0 = decimal.Decimal(4972)
        bigDecimal1 = bigDecimalWeightBaseOperations0.append(bigDecimal0, bigDecimal0)
        self.assertIsNotNone(bigDecimal1)

        int0 = bigDecimalWeightBaseOperations0.compare(bigDecimal0, bigDecimal1)
        self.assertEqual(-1, int0)

    def test08(self) -> None:

        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()
        bigDecimal0 = decimal.Decimal(-620)
        bigDecimal1 = bigDecimalWeightBaseOperations0.append(bigDecimal0, None)
        self.assertIsNone(bigDecimal1)

    def test07(self) -> None:

        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()
        try:
            bigDecimalWeightBaseOperations0.compare(None, None)
            self.fail("Expecting exception: RuntimeError")

        except Exception as e:
            self.verifyException(
                "org.apache.commons.graph.weight.primitive.BigDecimalWeightBaseOperations",
                e,
            )

    def test06(self) -> None:

        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()
        bigDecimal0 = decimal.Decimal(10)
        bigDecimal1 = bigDecimalWeightBaseOperations0.append(bigDecimal0, bigDecimal0)
        self.assertIsNot(bigDecimal1, bigDecimal0)

    def test05(self) -> None:

        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()
        bigDecimal0 = decimal.Decimal((-647))
        bigDecimal1 = bigDecimalWeightBaseOperations0.append(bigDecimal0, bigDecimal0)
        self.assertEqual(short((-1294)), bigDecimal1.to_eng_string())

    def test04(self) -> None:
        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()
        bigDecimal0 = bigDecimalWeightBaseOperations0.identity()
        bigDecimal1 = decimal.Decimal(-647)
        int0 = bigDecimalWeightBaseOperations0.compare(bigDecimal0, bigDecimal1)
        self.assertEqual(1, int0)

    def test03(self) -> None:
        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()
        bigDecimal0 = decimal.Decimal(-647)
        bigDecimal1 = bigDecimalWeightBaseOperations0.inverse(bigDecimal0)
        self.assertNotEqual(bigDecimal1, bigDecimal0)

    def test02(self) -> None:

        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()
        bigDecimal0 = decimal.Decimal(0)
        bigDecimal1 = decimal.Decimal(-1227)
        mathContext0 = decimal.Context(prec=128)
        bigDecimal2 = bigDecimal0.subtract(bigDecimal1, mathContext0)
        bigDecimal3 = bigDecimalWeightBaseOperations0.inverse(bigDecimal2)
        self.assertNotEqual(bigDecimal1, bigDecimal3)

    def test01(self) -> None:
        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()
        bigDecimal0 = bigDecimalWeightBaseOperations0.identity()
        bigDecimal1 = bigDecimalWeightBaseOperations0.inverse(bigDecimal0)
        self.assertEqual(0, bigDecimal1)

    def test00(self) -> None:

        bigDecimalWeightBaseOperations0 = BigDecimalWeightBaseOperations()
        bigDecimal0 = bigDecimalWeightBaseOperations0.identity()
        bigDecimal1 = decimal.Decimal(-647)
        bigDecimal2 = bigDecimalWeightBaseOperations0.append(bigDecimal0, bigDecimal1)
        self.assertEqual(short(-647), bigDecimal2.shortValue())

File: java_projects/cleaned_final_projects_evosuite/commons-graph/evosuite-tests/org/apache/commons/graph/weight/primitive/BigDecimalWeightBaseOperations_ESTest.java

/*
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */


package org.apache.commons.graph.weight.primitive;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.math.BigDecimal;
import java.math.MathContext;
import org.apache.commons.graph.weight.primitive.BigDecimalWeightBaseOperations;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class BigDecimalWeightBaseOperations_ESTest extends BigDecimalWeightBaseOperations_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      BigDecimal bigDecimal0 = bigDecimalWeightBaseOperations0.identity();
      BigDecimal bigDecimal1 = new BigDecimal((-647L));
      BigDecimal bigDecimal2 = bigDecimalWeightBaseOperations0.append(bigDecimal0, bigDecimal1);
      assertEquals((short) (-647), bigDecimal2.shortValue());
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      BigDecimal bigDecimal0 = bigDecimalWeightBaseOperations0.identity();
      BigDecimal bigDecimal1 = bigDecimalWeightBaseOperations0.inverse(bigDecimal0);
      assertEquals((byte)0, bigDecimal1.byteValue());
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      BigDecimal bigDecimal0 = BigDecimal.ZERO;
      BigDecimal bigDecimal1 = new BigDecimal((-1227L));
      MathContext mathContext0 = MathContext.DECIMAL128;
      BigDecimal bigDecimal2 = bigDecimal0.subtract(bigDecimal1, mathContext0);
      BigDecimal bigDecimal3 = bigDecimalWeightBaseOperations0.inverse(bigDecimal2);
      assertNotSame(bigDecimal1, bigDecimal3);
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      BigDecimal bigDecimal0 = new BigDecimal((-647L));
      BigDecimal bigDecimal1 = bigDecimalWeightBaseOperations0.inverse(bigDecimal0);
      assertNotSame(bigDecimal1, bigDecimal0);
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      BigDecimal bigDecimal0 = bigDecimalWeightBaseOperations0.identity();
      BigDecimal bigDecimal1 = new BigDecimal((-647L));
      int int0 = bigDecimalWeightBaseOperations0.compare(bigDecimal0, bigDecimal1);
      assertEquals(1, int0);
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      BigDecimal bigDecimal0 = new BigDecimal((-647L));
      BigDecimal bigDecimal1 = bigDecimalWeightBaseOperations0.append(bigDecimal0, bigDecimal0);
      assertEquals((short) (-1294), bigDecimal1.shortValue());
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      BigDecimal bigDecimal0 = BigDecimal.TEN;
      BigDecimal bigDecimal1 = bigDecimalWeightBaseOperations0.append(bigDecimal0, bigDecimal0);
      assertNotSame(bigDecimal1, bigDecimal0);
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      // Undeclared exception!
      try { 
        bigDecimalWeightBaseOperations0.compare((BigDecimal) null, (BigDecimal) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.weight.primitive.BigDecimalWeightBaseOperations", e);
      }
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      BigDecimal bigDecimal0 = new BigDecimal((-620));
      BigDecimal bigDecimal1 = bigDecimalWeightBaseOperations0.append(bigDecimal0, (BigDecimal) null);
      assertNull(bigDecimal1);
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      BigDecimal bigDecimal0 = new BigDecimal(4972);
      BigDecimal bigDecimal1 = bigDecimalWeightBaseOperations0.append(bigDecimal0, bigDecimal0);
      assertNotNull(bigDecimal1);
      
      int int0 = bigDecimalWeightBaseOperations0.compare(bigDecimal0, bigDecimal1);
      assertEquals((-1), int0);
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      BigDecimal bigDecimal0 = bigDecimalWeightBaseOperations0.append((BigDecimal) null, (BigDecimal) null);
      assertNull(bigDecimal0);
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      BigDecimal bigDecimal0 = bigDecimalWeightBaseOperations0.identity();
      BigDecimal bigDecimal1 = bigDecimalWeightBaseOperations0.append(bigDecimal0, bigDecimal0);
      assertSame(bigDecimal0, bigDecimal1);
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      // Undeclared exception!
      try { 
        bigDecimalWeightBaseOperations0.inverse((BigDecimal) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.weight.primitive.BigDecimalWeightBaseOperations", e);
      }
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      BigDecimalWeightBaseOperations bigDecimalWeightBaseOperations0 = new BigDecimalWeightBaseOperations();
      BigDecimal bigDecimal0 = BigDecimal.TEN;
      int int0 = bigDecimalWeightBaseOperations0.compare(bigDecimal0, bigDecimal0);
      assertEquals(0, int0);
  }
}
